cabal-install/Distribution/Client/Install.hs

```diff
@@ -280,9 +282,9 @@ makeInstallContext verbosity
     return (installedPkgIndex, sourcePkgDb, userTargets, pkgSpecifiers, transport)
 
 -- | Make an install plan given install context and install arguments.
-makeInstallPlan :: Verbosity -> InstallArgs -> InstallContext
-                -> IO (Progress String String InstallPlan)
-makeInstallPlan verbosity
+makeInstallPlans :: Verbosity -> InstallArgs -> InstallContext
```

Comment:
Its actually worse than this. I cannot just replace the function, if compiler does not support multiple instances (ghc < 7.10). But I was too frustrated of passing HcPkgInfo, Compiler, ProgramConfiguration to every function, so that I ignored it in some of the places to make the working product. I know this pull request is poorly written, but this was just for getting reviews. I can do remaining things latter.